cpp/include/ucxx/remote_key.h

```diff
@@ -0,0 +1,212 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <ucp/api/ucp.h>
+
+#include <ucxx/component.h>
+#include <ucxx/endpoint.h>
+#include <ucxx/memory_handle.h>
+
+namespace ucxx {
+
+typedef size_t SerializedRemoteKeyHash;
+
+/**
+ * @brief Component holding a UCP rkey (remote key).
+ *
+ * To provide RMA (Remote Memory Access) to memory handles, UCP packs their information
+ * in the form of `ucp_rkey_h` (remote key, or rkey for short). This class encapsulates
+ * that object and provides methods to simplify its handling, both locally and remotely
+ * including (de-)serialization for transfers over the wire and reconstruction of the
+ * object on the remote process.
+ */
+class RemoteKey : public Component {
+ private:
+  ucp_rkey_h _remoteKey{nullptr};  ///< The unpacked remote key.
+  void* _packedRemoteKey{
+    nullptr};                      ///< The packed (local) remote key that may be unpacked (remote).
```

Comment:
*question*: I think I can't quite parse this docstring. Is this the key that we generated locally, packed to send to the remote?

Reply:
Perhaps "The packed `ucp_rkey_h` key, suitable for transfer to a remote" ?

Reply:
Agreed, this is a better phrasing. Done via https://github.com/rapidsai/ucxx/pull/190/commits/a700b11efa50673b152c4742edc5fbc562d43bc4